src/audio/audio_context.rs

```diff
@@ -0,0 +1,37 @@
+//! Provides a context for loading a file.
+
+use super::source::Source;
+use assets::*;
+
+use std::sync::Arc;
+
+/// A context for loading audio files
+pub struct AudioContext {
+    inner: SimpleContext<Source, Vec<u8>, NoError, fn(Vec<u8>) -> Result<Source, NoError>>,
```

Comment:
Not sure about that dynamic dispatch here..I think we should replace that. Pretty sure a change in `amethyst_assets` is required.